tests/Catalog.FunctionalTests/CatalogApiTests.cs

```diff
@@ -87,4 +87,224 @@ public async Task UpdateCatalogItemWorksWithPriceUpdate()
+        var body = await response.Content.ReadAsStringAsync();
+        var result = JsonSerializer.Deserialize<PaginatedItems<CatalogItem>>(body, _jsonSerializerOptions);
+
+        // Assert       
+        Assert.Equal(1, result.Count);
+        Assert.NotNull(result.Data);
+        Assert.Equal(0, result.PageIndex);
+        Assert.Equal(5, result.PageSize);
+    }
+
+    [Fact]
+    public async Task GetCatalogItemWithTypeIdBrandId()
+    {
+        // Act
+        var response = await _httpClient.GetAsync("api/v1/catalog/items/type/3/brand/3?PageSize=5&PageIndex=0");
+
+        // Arrange   
+        response.EnsureSuccessStatusCode();
+        var body = await response.Content.ReadAsStringAsync();
+        var result = JsonSerializer.Deserialize<PaginatedItems<CatalogItem>>(body, _jsonSerializerOptions);
+
+        // Assert    
+        Assert.NotNull(result.Data);
+        Assert.Equal(4, result.Count);
+        Assert.Equal(0, result.PageIndex);
+        Assert.Equal(5, result.PageSize);
+        Assert.Equal(3, result.Data.ToList().FirstOrDefault().CatalogTypeId);
+        Assert.Equal(3, result.Data.ToList().FirstOrDefault().CatalogBrandId);
+    }
+
+    [Fact]
+    public async Task GetAllCatalogTypeItemWithBrandId()
+    {
+        // Act
+        var response = await _httpClient.GetAsync("api/v1/catalog/items/type/all/brand/3?PageSize=5&PageIndex=0");
+
+        // Arrange
+        response.EnsureSuccessStatusCode();
+        var body = await response.Content.ReadAsStringAsync();
+        var result = JsonSerializer.Deserialize<PaginatedItems<CatalogItem>>(body, _jsonSerializerOptions);
+
+        // Assert              
+        Assert.NotNull(result.Data);
+        Assert.Equal(11, result.Count);
+        Assert.Equal(0, result.PageIndex);
+        Assert.Equal(5, result.PageSize);
+        Assert.Equal(3, result.Data.ToList().FirstOrDefault().CatalogBrandId);
+    }
+
+    [Fact]
+    public async Task GetAllCatalogTypes()
+    {
+        // Act
+        var response = await _httpClient.GetAsync("api/v1/catalog/catalogtypes");
+
+        // Arrange   
+        response.EnsureSuccessStatusCode();
+        var body = await response.Content.ReadAsStringAsync();
+        var result = JsonSerializer.Deserialize<List<CatalogType>>(body, _jsonSerializerOptions);
+
+        // Assert       
+        Assert.Equal(8, result.Count);
+        Assert.NotNull(result);
+    }
+
+    [Fact]
+    public async Task GetAllCatalogBrands()
+    {
+        // Act
+        var response = await _httpClient.GetAsync("api/v1/catalog/catalogbrands");
+
+        // Arrange   
+        response.EnsureSuccessStatusCode();
+        var body = await response.Content.ReadAsStringAsync();
+        var result = JsonSerializer.Deserialize<List<CatalogBrand>>(body, _jsonSerializerOptions);
+
+        // Assert       
+        Assert.Equal(13, result.Count);
+        Assert.NotNull(result);
+    }
+
+    [Fact]
+    public async Task AddDeleteCatalogItem()
```

Comment:
Can we split these up into two tests? One for adding and one for deleting.

Reply:
I can do that. 
The reason of testing in same test case is:  If adding to the particular catalog item(which is getting deleted in delete test) in post method will fail then Delete first assert statement will fails as it will return "NotFound" instead of "NoContent".
Also, if we run delete test case before post test case it will fail as I am deleting the catalog item added in Post test.
Considering that in mind I wrote both functionalities testing together.

Reply:
Yeah, I'm fine writing one test case that exclusively tests post and then another that tests delete. There's no strong requirement to need to insert a new element in order to delete it since we setup the test DB from for each run.

Reply:
@captainsafia Resolving this based on your comment. Thanks